Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -140,3 +217,26 @@ public struct OfflineMapAreasView: View {
     }
     return OfflineMapAreasViewPreview()
 }
+
+@MainActor
+private class NetworkMonitor: ObservableObject {
+    /// The path monitor to observe network changes.
+    private let monitor = NWPathMonitor()
+    
+    /// A Boolean value indicating whether the device has an internet connection.
+    @Published private(set) var isConnected = true
+    
+    init() {
+        monitor.pathUpdateHandler = { [unowned self] path in
+            DispatchQueue.main.async {
+                self.isConnected = path.status == .satisfied
+            }
+        }
+        let queue = DispatchQueue(label: "NetworkMonitor")
+        monitor.start(queue: queue)
+    }
```

Comment:
In this case, we can maybe conform the `NetworkMonitor` to `Cancellable` protocol as well?

Reply:
Uh, no. `Cancellable` is provided by `Combine`, which isn't even being used here. How would that be beneficial?

Reply:
Dang… I thought it was Swift Concurrency. NVM

Reply:
After implementing this suggestion I realized it introduced a timing bug when the device is fully offline. Since `isConnected` is set to `true` by default, the online models are created before the network connection is set by the network monitor so an error message is presented for no internet connection. This issue is fixed by the latest commit [cbbaf44](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/866/commits/cbbaf444163c99c99662675240a14f323fb0d517).